common/environment/environment.cpp

```diff
@@ -3215,3 +3215,45 @@ unsigned getEnvironmentThorClusterNames(StringArray &thorNames, StringArray &gro
     }
     return thorNames.ordinality();
 }
+
+IConstWUClusterInfo *getContainerTargetClusterInfo(IPropertyTree *queue)
+{
+    IArrayOf<IPropertyTree> thors;
+    Owned<IPropertyTree> roxie = nullptr;
+    const char *name = queue->queryProp("@name");
+    const char *type = queue->queryProp("@type");
+    if (strieq(type, "thor"))
+    {
+        Owned<IPropertyTree> thor = createPTree(name, ipt_caseInsensitive);
+        thors.append(*thor.getClear());
+    }
+    else if (strieq(type, "roxie"))
+    {
+        roxie.setown(createPTree(name, ipt_caseInsensitive));
+    }
+
+    IArrayOf<IPropertyTree> eclServers;
```

Comment:
how are the eclccserver picked up show in container mode?

Reply:
Good question. I haven't found out a way to pick up eclccserver in container mode.

Reply:
It depends what we want the view to look like.
The topology is different from a legacy system, where a target defined the eclccserver (and agent and thor etc.) that were to be associated with it.

The primary view/info for the activity page we probably want, is to see the targets (the hthor, thor and roxie queues), and the state of the workunits that have been submitted to them.
Each target queue, has several queues associated with it. e.g. a thor has an compile queue (eclccserver), a workflow queue (eclagent), and a thor agent queue (that actually spins up thor instances).
Knowing the state of a workunit, as it transitions through these queues/other states is useful, it's secondary information, that should probably be accessed or linked to under the main list of active workunits under each target.

I think the esp activity service should return a list of the queue targets, with the workunits associated with them, and some state information.